lxd/devlxd.go

```diff
@@ -487,8 +489,24 @@ func registerDevLXDEndpoint(d *Daemon, apiRouter *mux.Router, apiVersion string,
 
 	// Function that handles the request by calling the appropriate handler.
 	handleFunc := func(w http.ResponseWriter, r *http.Request) {
-		// Set devLXD auth method to identify this request as coming from the /dev/lxd socket.
-		err := request.SetRequestor(r, d.identityCache, request.RequestorArgs{Protocol: request.ProtocolDevLXD})
+		var requestor request.RequestorArgs
+
+		// Set [request.ProtocolDevLXD] by default identify this request as coming from the /dev/lxd socket.
+		requestor.Protocol = request.ProtocolDevLXD
+
+		// Check if the caller has a bearer token.
+		isBearerRequest, token, subject := bearer.IsRequest(r, d.globalConfig.ClusterUUID())
+		if isBearerRequest {
+			bearerRequestor, err := bearer.Authenticate(token, subject, d.identityCache, api.IdentityTypeBearerTokenDevLXD)
+			if err == nil {
+				requestor = *bearerRequestor
+			} else if !auth.IsDeniedError(err) {
+				// Log any internal errors
+				logger.Warn("Failed to authenticate DevLXD bearer token", logger.Ctx{"err": err})
```

Comment:
Just return an error. Easy to relax restriction but hard to restrict if a problem

Reply:
@tomponline and I discussed this earlier. We said we'd reject the request if a token is sent that is not valid, regardless or the error. I didn't do this for the reason you have mentioned above

Reply:
This is now changed to always reject invalid tokens, regardless of the endpoint they're hitting. We can relax this restriction later if required.